### PR TITLE
Added the name of the BlockArray class to __repr__

### PR DESCRIPTION
**Pull request by FernandoDavis**
Added the name of the BlockArray class to __repr__

### CODE REVIEW

File: scico/numpy/_blockarray.py [hunk 1]
```diff
@@ -79,6 +79,9 @@ def blockarray(iterable):
         """Construct a :class:`.BlockArray` from a list or tuple of existing array-like."""
         return BlockArray(iterable)
 
+    def __repr__(self):
+        return f"snp.BlockArray({super().__repr__()})"
```
Comment (Michael-T-McCann) [inline scico/numpy/_blockarray.py:83]: @bwohlberg How about just `BlockArray(...` rather than `snp.BlockArray(...`? The former would match `DeviceArray`.

Comment (bwohlberg) [reply scico/numpy/_blockarray.py:83]: Agreed.

**Comment by Michael-T-McCann** [comment]
@bwohlberg Any ideas on these check failures?

**Comment by bwohlberg** [comment]
I'm hoping the errors are transitory.

**Comment by bwohlberg** [comment]
Looks good now.

**Comment by bwohlberg** [comment]
@Michael-T-McCann: The current output of the code in this branch is not quite as outlined in #237. Did you envisage stripping the `DeviceArray` as in the example there, or leaving it in place as in the output for this branch?

**Comment by Michael-T-McCann** [comment]
> Did you envisage stripping the `DeviceArray`

I did, but seeing how it looks I'm no longer sure. `BlockArray`s should only ever contain `DeviceArray`s so having "DeviceArray" in the repr is redundant. However, we do hope to support heterogeneous data types and having those types displayed in the repr would be useful. Removing "DeviceArray" but leaving the dtype looks odd to me:

```
    BlockArray([([[ 2,  6, 14],
		 [ 4,  4,  2]], dtype=...), ([ 4,  8, 16], dtype=...)])
```

I'm inclined to merge this as is.

**Pull request merged**